Automattic/sensei
# Api for removing multiple users from multiple courses

Fixes part of https://github.com/Automattic/sensei/issues/4959

### Changes proposed in this Pull Request
Add REST API endpoint to add users to courses

### Testing instructions
- Create a few users and a few courses

- Enroll those users to those courses, either from the UI or by using the rest API here https://github.com/Automattic/sensei/pull/4968

- Get WP REST nonce (make sure you're logged in): http://devwp.local/wp-admin/admin-ajax.php?action=rest-nonce (replace devwp.local with your domain)

- Copy Cookie name and value
- Execute in terminal (replace WPNONCE, COOKIE-NAME & COOKIE-VALUE, USER_ID, COURSE_ID with actual values):
```
curl -X "DELETE" "http://devwp.local/?rest_route=%2Fsensei-internal%2Fv1%2Fcourse-students%2Fbatch&_wpnonce=WPNONCE" \
     -H 'Cookie: COOKIE-NAME=COOKIE-VALUE' \
     -H 'Content-Type: application/json; charset=utf-8' \
     -d $'{
  "student_ids": USER_ID,
  "course_ids": COURSE_ID
}'
```
Make sure you get status code 200 and your users are removed from the courses.

If you use vscode's thunder client, you can import the following collection for testing the API. Change the URL and the variables as per your system config
[thunder-collection_Sensei.json.zip](https://github.com/Automattic/sensei/files/8428461/thunder-collection_Sensei.json.zip)
